### ISSUE TITLE
Error when copying file that contains a semicolon ";"

### ISSUE THREAD

**Issue by antonhagg** [rclone/rclone#2750]
Not sure if this has been covered yet, but when syncing from Jottacloud to winows, files that contain a semicolon ";" get skipped.

#### What is your rclone version (output from `rclone version`)
v1.44-097-g2191592e-beta

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Windows 64 Bit

####  Which cloud storage system are you using? (eg Google Drive)
JottaCloud

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
2018/11/20 01:05:07 DEBUG : rclone: Version "v1.44-097-g2191592e-beta" starting with parameters ["rclone" "copy" "jottacloud:hejsan\\testfolder" "e:\\rclone-v1.44\\hejsan" "--checksum" "--progress" "--log-file" "hejsan.txt" "--log-level" "DEBUG"]
2018/11/20 01:05:07 DEBUG : Using config file from "C:\\Users\\Ant\\.config\\rclone\\rclone.conf"
2018/11/20 01:05:08 INFO  : Local file system at \\?\e:\rclone-v1.44\hejsan: Waiting for checks to finish
2018/11/20 01:05:08 INFO  : Local file system at \\?\e:\rclone-v1.44\hejsan: Waiting for transfers to finish
**_2018/11/20 01:05:09 ERROR : Jottacloud;withsemicolon.pdf: Failed to copy: failed to open source object: error 404: no.jotta.backup.errors.NoSuchPathException (Not Found)_**

**Comment by albertony** [comment]
I have experienced the same, and mentioned it in the discussion of closed issue #2531. It is intentional. Some quotes:

>The problem with ; is that the web ui will just eat it and all following characters even though this is probably a bug in the web ui I decided to keep replacing it.


>Ah, I can see that behavior now - when attempting to create a new folder using the web ui. But creating it with rclone works fine (when I comment out the replacing of it), and then web ui can work with it without issues as far as I can see. And uploading a file with ; in it works fine from web ui.

So my suggestion was, and still is, to open up for ; in names the Jottacloud backend.

**Comment by ncw** [comment]
I managed to replicate this by uploading a file with `;` via the web interface then downloading it via rclone.

When you upload a file with `;` with rclone replaces `;` with a unicode eqiuvalent `；`.

@albertony is right though - the web interface hates files with `;` in - try renaming a file with one in to see what happens!

> So my suggestion was, and still is, to open up for ; in names the Jottacloud backend.

@B4dM4n is working on tidying this up for all backends at the moment in #2633

I took the character escaping out and then ran

```
$ rclone info jottacloud:testchars
charNeedsEscaping = []byte{
	0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F, 0x22, 0x2A, 0x2F, 0x3A, 0x3C, 0x3E, 0x3F, 0x7C, 0x7F
}
```

So it looks like that jottacloud can't store any control characters 0x00-0x1F,0x7F  or `['"', '*', '/', ':', '<', '>', '?', '|']`

Currently we are escaping

```go
		'\\': '＼', // FULLWIDTH REVERSE SOLIDUS
		'*':  '＊', // FULLWIDTH ASTERISK
		'<':  '＜', // FULLWIDTH LESS-THAN SIGN
		'>':  '＞', // FULLWIDTH GREATER-THAN SIGN
		'?':  '？', // FULLWIDTH QUESTION MARK
		':':  '：', // FULLWIDTH COLON
		';':  '；', // FULLWIDTH SEMICOLON
		'|':  '｜', // FULLWIDTH VERTICAL LINE
		'"':  '＂', // FULLWIDTH QUOTATION MARK - not on the list but seems to be reserved
```

So it looks like these ones aren't necessary

```go
		'\\': '＼', // FULLWIDTH REVERSE SOLIDUS
		';':  '；', // FULLWIDTH SEMICOLON
```

I'd be happy to remove these from the replacments even if they don't work well in the web interface, as long as they are 100% via rclone.

Thoughts? @buengese 

Are there any definitive docs from Jottacloud on this?

**Comment by buengese** [comment]
I’m aware of everything discussed in here. It was an intentional decision to keep replacing ‘;’ to prevent users from getting themselves into trouble with the webui while ‘:’ does in fact relate to the problem #2633. I’am however very surprised how many people use special characters in there file names and I’m open to change this behavior. For consistency reasons the behavior regarding spaces as reported in #2744 should be changed to.

> Are there any definitive docs from Jottacloud on this?

Obviously not (and I don’t really think they even keep track of these internally given how broken this is)

**Comment by hajes** [comment]
with semicolon in file name...you will get problem on any POSIX system...or systems using ASCII only characters...use simple names without any special characters and you can sleep in peace.

**Comment by antonhagg** [comment]
@hajes I Would never think of using semicolon in filenames, but when doing a full backup of a machine I can't have control of what others use as filenames. This specific case is Skype who have used semicolon in some of their filenames.

**Comment by jjgoaway** [comment]
Hello fine people. Just dropping in to see if there was an update on this. I came across this issue and had some quick conversation on the forum and was directed here.
https://forum.rclone.org/t/files-with-seem-to-be-having-issues/9694/4

**Comment by ncw** [comment]
I've merged the fix for this to master now which means it is in [the latest beta](https://beta.rclone.org) and will be released in v1.50
